fixieai/agents-ts/src/template.ts

```diff
@@ -9,26 +9,27 @@
  *     https://github.com/fixie-ai/fixie-examples
  */
 
-// @ts-expect-error
-import { CodeShotAgent, Message } from 'fixieai';
+export const BASE_PROMPT = `I'm an agent that rolls virtual dice!`;
 
-const BASE_PROMPT = 'General info about what this agent does and the tone it should use.';
+export const FEW_SHOTS = `
+Q: Roll a d20
+Ask Func[roll]: 20 1
+Func[roll] says: 12
+A: You rolled a 12!
 
-const FEW_SHOTS = `
-Q: Sample query to this agent
-A: Sample response
+Q: Roll two dice and blow on them first for good luck
+Ask Func[roll]: 6 2
+Func[roll] says: 4 3
+A: You rolled a 4 and a 3, with a total of 7.
 
-Q: Another sample query
-Ask Func[example]: input
-Func[example] says: output
-A: The other response is output
+Q: Roll 3d8
+Ask Func[roll]: 8 3
+Func[roll] says: 5 3 8
+A: You rolled 5, 3, and 8, for a total of 16.
 `;
 
-const agent = new CodeShotAgent(BASE_PROMPT, FEW_SHOTS);
-
-agent.registerFunc((query: Message): string => {
-  if (query.text === 'input') {
-    return 'output';
-  }
-  throw new Error('Invalid input');
-});
+export function roll(query: { text: string; }): string {
```

Comment:
I like the module-as-agent paradigm!

Since there's a handful of other config options as well, I'm wondering if we should wrap all the config in a single object, like `const export config = { base_prompt: "...", few_shots: "...", conversational: true }`, WDYT?

The other thing to consider is how users would configure a standalone agent, but I think mirroring the `GET /` response with a `"type"` field seems probably sufficient?

Reply:
Discussed in person + [Slack](https://fixie-ai.slack.com/archives/C03H4FX54JZ/p1680803400136669). We'll defer making a change on this.